src/MTGCore/Services/SearchFilterMapper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using MTGCore.Services.Exceptions;
+using MTGCore.ViewModels;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace MTGCore.Services
+{
+    public class SearchFilterMapper : ISearchFilterMapper
+    {
+        private Dictionary<string, string> filterParameters = new Dictionary<string, string>();
+
+        private Dictionary<Type, Delegate> typeProcessorMap = new Dictionary<Type, Delegate>() 
+        {
+            { typeof(List<string>), new Func<PropertyInfo,SearchFilter,string>((x,y) => {return StringListResolver.ValueToString(x,y);}) },
+            { typeof(string), new Func<PropertyInfo,SearchFilter,string>((x,y) => {return StringResolver.ValueToString(x,y);}) }
+        };
+
+        public Dictionary<string, string> map(SearchFilter filter)
+        {
+
+            PropertyInfo[] properties = typeof(SearchFilterWithColours).GetProperties();
+            foreach (PropertyInfo property in properties)
```

Comment:
**Suggestion:** I quite like the style of moving this logic into a private method with the params `PropertyInfo` and `SearchFilter`.

![image](https://user-images.githubusercontent.com/17239606/119729091-26b3e000-be6c-11eb-8d08-e444ac0a3b00.png)

You could then use Linq to pass each value in and either return the key and value or add it directly in the method.

![image](https://user-images.githubusercontent.com/17239606/119729177-4519db80-be6c-11eb-8c04-d70d65450a8e.png)

It depends which way you think is easier to understand.